orchagent/pfc_detect_mlnx.lua

```diff
@@ -0,0 +1,86 @@
+
+-- Iterate through each queue
+local n = table.getn(KEYS)
+for i = n, 1, -1 do
+    local counter_keys = redis.call('HKEYS', counters_table_name .. ':' .. KEYS[i])
+    local counter_num = 0
+    local old_counter_num = 0
+    local is_deadlock = false
+    local pfc_wd_status = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'PFC_WD_STATUS')
+    if pfc_wd_status == 'operational' then
+        local detection_time = tonumber(redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'PFC_WD_DETECTION_TIME'))
+        local time_left = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'PFC_WD_DETECTION_TIME_LEFT')
+        if not time_left  then
+            time_left = detection_time
+        else
+            time_left = tonumber(time_left)
+        end
+
+        local queue_index = redis.call('HGET', 'COUNTERS_QUEUE_INDEX_MAP', KEYS[i])
+        local port_id = redis.call('HGET', 'COUNTERS_QUEUE_PORT_MAP', KEYS[i])
+        local pfc_rx_pkt_key = 'SAI_PORT_STAT_PFC_' .. queue_index .. '_RX_PKTS'
+        local pfc_duration_key = 'SAI_PORT_STAT_PFC_' .. queue_index .. '_RX_PAUSE_DURATION'
+
+        -- Get all counters
+        local occupancy_bytes = tonumber(redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'SAI_QUEUE_STAT_CURR_OCCUPANCY_BYTES'))
+        local packets = tonumber(redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'SAI_QUEUE_STAT_PACKETS'))
+        local pfc_rx_packets = tonumber(redis.call('HGET', counters_table_name .. ':' .. port_id, pfc_rx_pkt_key))
+        local pfc_duration = tonumber(redis.call('HGET', counters_table_name .. ':' .. port_id, pfc_duration_key))
+
+        local packets_last = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'SAI_QUEUE_STAT_PACKETS_last')
+        local pfc_rx_packets_last = redis.call('HGET', counters_table_name .. ':' .. port_id, pfc_rx_pkt_key .. '_last')
+        local pfc_duration_last = redis.call('HGET', counters_table_name .. ':' .. port_id, pfc_duration_key .. '_last')
+        -- DEBUG CODE START. Uncomment to enable
+        local debug_storm = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'DEBUG_STORM')
+        -- DEBUG CODE END.
+
+        -- If this is not a first run, then we have last values available
+        if packets_last and pfc_rx_packets_last and pfc_duration_last then
+            packets_last = tonumber(packets_last)
+            pfc_rx_packets_last = tonumber(pfc_rx_packets_last)
+            pfc_duration_last = tonumber(pfc_duration_last)
+
+            -- Check actual condition of queue being in PFC storm
+            if (occupancy_bytes > 0 and packets - packets_last == 0 and pfc_rx_packets - pfc_rx_packets_last > 0) or
+                -- DEBUG CODE START. Uncomment to enable
+                (debug_storm == "enabled") or
+                -- DEBUG CODE END.
+                (occupancy_bytes == 0 and packets - packets_last == 0 and (pfc_duration - pfc_duration_last) > poll_time * 0.8) then
```

Comment:
@marian-pritsak Looks like the value of 'SAI_PORT_STAT_PFC_#_RX_PAUSE_DURATION' return from get_stat is in milliseconds as it is directly compared with poll_time, which is 200 millisecond. Does it comply with the SAI definition of 'SAI_PORT_STAT_PFC_#_RX_PAUSE_DURATION'